structuralcodes/sections/_generic.py

```diff
@@ -96,9 +100,135 @@ def _calculate_gross_section_properties(self) -> s_res.GrossProperties:
         """
         # It will use the algorithms for generic sections
         gp = s_res.GrossProperties()
-        # Here compute area, centroid and other stuff from gross properties
-        # 1. Computation of area
+
+        # Computation of perimeter using shapely
+        polygon = unary_union(
+            [geo.polygon for geo in self.section.geometry.geometries]
+        )
+        if isinstance(polygon, MultiPolygon):
+            gp.perimeter = 0.0
+            warnings.warn(
+                'Perimiter computation for a multi polygon is not defined.'
+            )
+
+        gp.perimeter = polygon.exterior.length
+
+        # Computation of area: this is taken directly from shapely
         gp.area = self.section.geometry.area
+        # Computation of surface area, reinforcement area, EA (axial rigidity)
+        # and mass: Morten -> problem with units! how do we deal with it?
```

Comment:
Good question. We specify in docstrings that the density should be given in kg/m3, but we are not specifying any unit for the geometry. I think it is ok for now, but we should do a more thorough review of all connected parts of the code before v0.0.1.